cpp/src/arrow/engine/substrait/expression_internal.cc

```diff
@@ -710,6 +710,15 @@ struct ScalarToProtoImpl {
                       s);
   }
 
+  Status Visit(const StringViewScalar& s) {
```

Comment:
Substrait changes can probably left for another PR (I suppose you'll need to write tests for them?).